pkg/server/sessions/manager.go

```diff
@@ -65,30 +63,38 @@ func NewManager(options *Options) (*manager, error) {
 	if options == nil {
 		return nil, ErrInvalidOptionsProvided
 	}
-	if options.MaxSessionAgeTime == 0 {
-		options.MaxSessionAgeTime = infinity
-	}
-	if options.MaxSessionInactivityTime == 0 {
-		options.MaxSessionInactivityTime = infinity
-	}
-	if options.Timeout == 0 {
-		options.Timeout = infinity
+
+	err := options.Validate()
+	if err != nil {
+		return nil, err
 	}
+
 	guard := &manager{
 		sessions: make(map[string]*Session),
 		ticker:   time.NewTicker(options.SessionGuardCheckInterval),
 		done:     make(chan bool),
 		logger:   logger.NewSimpleLogger("immudb session guard", os.Stdout),
-		options:  options,
+		options:  *options,
+	}
+
+	if guard.options.MaxSessionAgeTime == 0 {
+		guard.options.MaxSessionAgeTime = infinity
+	}
+	if guard.options.MaxSessionInactivityTime == 0 {
+		guard.options.MaxSessionInactivityTime = infinity
 	}
+	if guard.options.Timeout == 0 {
+		guard.options.Timeout = infinity
+	}
+
 	return guard, nil
 }
 
 func (sm *manager) NewSession(user *auth.User, db database.DB) (*Session, error) {
 	sm.sessionMux.Lock()
 	defer sm.sessionMux.Unlock()
 
-	if len(sm.sessions) == MaxSessions {
+	if len(sm.sessions) >= sm.options.MaxSessions {
```

Comment:
when was the case it could be > ?

Reply:
Hmm, currently it should not be possible due to options validation, but it would happen if someone set MaxSessions option to something negative - in such case you have unbound session count